### ISSUE TITLE
UnstructuredMarkdownLoader resulting in `zipfile.BadZipFile: File is not a zip file`

### ISSUE THREAD

**Issue by abhishekbhakat** [langchain-ai/langchain#5264]
### System Info

```
langchain==0.0.180
Python 3.11.3 (tags/v3.11.3:f3909b8, Apr  4 2023, 23:49:59) [MSC v.1934 64 bit (AMD64)] on win32
Windows 11
```

### Who can help?

@eyurtsev

### Information

- [X] The official example notebooks/scripts
- [ ] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [X] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

Python Code
```
from langchain.document_loaders import UnstructuredMarkdownLoader
markdown_path = r"Pyspy.md"
loader = UnstructuredMarkdownLoader(markdown_path)
data = loader.load()
```

Markdown file `Pyspy.md`
```
    ```
     .pip/bin/py-spy top -p 70
    ```
```

### Expected behavior

It should result in List[Document] in data

**Comment by Friedrich-hue** [comment]
same problem

**Comment by devilteo911** [comment]
I have the same problem with `UnstructuredPDFLoader`. As mentioned by @abhishekbhakat, I was using privateGPT too but then I switched to a notebook to isolate the problem. 
Just to mention, I have the (currently) last version of langchain (0.0.250) and unstructured["all-docs"] (0.9)

Here it is the code:

```python
from langchain.document_loaders.pdf import UnstructuredPDFLoader
pdf = UnstructuredPDFLoader("source_documents/2212.10496.pdf", mode="single")
pdf.load()
```

The pdf link to replicate: [Paper](https://arxiv.org/pdf/2212.10496.pdf)

this is the full traceback:
```bash
BadZipFile                                Traceback (most recent call last)
Cell In[14], line 1
----> 1 pdf.load()

File /White2TB/morom/privateGPT/venv/lib/python3.10/site-packages/langchain/document_loaders/unstructured.py:86, in UnstructuredBaseLoader.load(self)
     84 def load(self) -> List[Document]:
     85     """Load file."""
---> 86     elements = self._get_elements()
     87     if self.mode == "elements":
     88         docs: List[Document] = list()

File /White2TB/morom/privateGPT/venv/lib/python3.10/site-packages/langchain/document_loaders/pdf.py:57, in UnstructuredPDFLoader._get_elements(self)
     54 def _get_elements(self) -> List:
     55     from unstructured.partition.pdf import partition_pdf
---> 57     return partition_pdf(filename=self.file_path, **self.unstructured_kwargs)

File /White2TB/morom/privateGPT/venv/lib/python3.10/site-packages/unstructured/documents/elements.py:222, in process_metadata..decorator..wrapper(*args, **kwargs)
    220 @wraps(func)
    221 def wrapper(*args, **kwargs):
--> 222     elements = func(*args, **kwargs)
    223     sig = inspect.signature(func)
    224     params = dict(**dict(zip(sig.parameters, args)), **kwargs)

File /White2TB/morom/privateGPT/venv/lib/python3.10/site-packages/unstructured/file_utils/filetype.py:628, in add_metadata_with_filetype..decorator..wrapper(*args, **kwargs)
...
-> 1336     raise BadZipFile("File is not a zip file")
   1337 if self.debug > 1:
   1338     print(endrec)

BadZipFile: File is not a zip file
```

**Comment by agombert** [comment]
Same problem as @devilteo911

**Comment by 18811449050** [comment]
I have same problem Do you solved the problem?

**Comment by devilteo911** [comment]
Hey guys,

today I've been able to solve my problem. I will give you the steps I've taken and hopefully everything will work for you too. I'm using the current latest release of unstructured (0.10.9)

1. Install the [system dependency](https://github.com/Unstructured-IO/unstructured#installing-the-library) requirements.
2. I saw that the `BadZipFile: File is not a zip file` was due to a problem with `nltk`. In fact, the following function wasn't able to download and handle the zip: 
```python
def _download_nltk_package_if_not_present(package_name: str, package_category: str):
    """If the required nlt package is not present, download it."""
    try:
        nltk.find(f"{package_category}/{package_name}")
    except LookupError:
        nltk.download(package_name)
```
it can be found here: `venv/lib/python3.xx/unstructured/nlp/tokenize.py`
3. To solve, I manually installed the missing nltk library by doing (in a separated script or directly on the python shell):
```python
import nltk
for pkg in ['punkt', 'averaged_perceptron_tagger']:
   nltk.download(pkg)
```

Now I was able to use the unstructured library without any error! 👍🏻 (Let me know if this works for you @agombert and @18811449050)
